packages/react-components/src/components/ErrorBar.tsx

```diff
@@ -123,29 +128,115 @@ export interface ActiveError {
  * All strings that can be shown are accepted as the {@link ErrorBarProps.strings} so that they can be localized.
  * Active errors are selected by {@link ErrorBarProps.activeErrors}.
  *
+ * This component internally tracks dismissed by the user.
+ *   * Errors that have an associated timestamp: The error is shown on the UI again if it occurs after being dismissed.
+ *   * Errors that do not have a timestamp: The error is dismissed until it dissappears from the props.
+ *         If the error recurs, it is shown in the UI.
+ *
  * Uses {@link @fluentui/react#MessageBar} UI element.
  */
 export const ErrorBar = (props: ErrorBarProps): JSX.Element => {
   const localeStrings = useLocale().strings.errorBar;
   const strings = props.strings ?? localeStrings;
 
-  if (props.activeErrors.length === 0) {
-    return <></>;
-  }
+  const [dismissedErrors, setDismissedErrors] = useState<DismissedError[]>([]);
+
+  // dropDismissalsForInactiveErrors only returns a new object if `dismissedErrors` actually changes.
+  // Without this behaviour, this `useEffect` block would cause a render loop.
+  useEffect(
+    () => setDismissedErrors(dropDismissalsForInactiveErrors(props.activeErrors, dismissedErrors)),
+    [props.activeErrors, dismissedErrors]
+  );
+
+  const toShow = errorsToShow(props.activeErrors, dismissedErrors);
 
-  // FIXME: Memoize onDismiss callbacks.
   return (
     <Stack>
-      {props.activeErrors.map((activeError) => (
+      {toShow.map((activeError) => (
```

Comment:
Rename 'activeError' to 'errorToShow', I thought this was an instance of props.activeErrors at the first glance

Reply:
Renamed to just `error`. The name `errorsToShow` is already taken by a helper function.